src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -722,13 +723,15 @@ private void OnAnalyzerChanged(object sender, EventArgs e)
             // efforts to listen to file changes can lead to a deadlock situation.
             // Postponing the VisualStudioAnalyzer operations gives this thread the opportunity
             // to release the lock.
-            Dispatcher.CurrentDispatcher.BeginInvoke(new Action(() =>
+            ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
             {
-                VisualStudioAnalyzer analyzer = (VisualStudioAnalyzer)sender;
+                await Task.Yield();
```

Comment:
ConfigureAwait(true)? (how did this get past the analyzer?)

Reply:
➡️ `Task.Yield()` returns `YieldAwaitable`, a custom struct awaitable.